dashboard/app/models/user.rb

```diff
@@ -237,6 +241,20 @@ def admins_must_be_teachers_without_followeds
     end
   end
 
+  def email
+    unless provider && provider == 'migrated'
+      return self[:email]
```

Comment:
Might be more clear to use `read_attribute(:email)`

nit: ruby-style one liner: 
```ruby
return read_attribute(:email) unless provider == 'migrated'
```